File: solutions/ruby/hamming/2/hamming.rb
class Hamming
  def self.compute(str1, str2)
    raise ArgumentError if str1.length != str2.length
    distance = 0
    (0...str1.length).each do |i|
      distance += 1 if str1[i] != str2[i]
    end
    distance
  end
end

File: solutions/c/hamming/2/hamming.c
#include "hamming.h"
#include <string.h>

int compute(const char *lhs, const char *rhs)
{
    size_t i = 0;
    int d = 0;
    while (lhs[i] && rhs[i])
    {
        d += lhs[i] != rhs[i];
        i++;
    }
    return (lhs[i] || rhs[i]) ? -1 : d;
}